Improve devcontainer and update to Python 3.12, add Azure CLI, postCr…

iMicknl opened the pull request:
@chuwik did create my devcontainer already, and just saw your recent update! Decided to add my changes to your devcontainer. Let me know if the changes make any sense and if you are open for contributoin, or if you would like any of the changes reverted.

- Added default comments to devcontainer to make it easier for first time users
- Bump devcontainer to use Python 3.12 (as supported by the project)
- Added Copilot (Chat) as default extension
- Added Azure CLI (required for auth via Azure Identity support instead of key based access)
- Added postCreateCommand to install the dependencies on container creation
- Remove vscode user, which is not required with these default devcontainer images
- Add devcontainer to dependabot updates

chuwik commented:
Thanks @iMicknl! 

Couple of suggestions:
- The project now supports python 3.11, let's keep the image on the lower version.
- Would it better to run ./start.sh in postCreateCommand rather than just restoring the packages? That'd restore + build + run the app straight away. If you can test it and it works, I'd say go for it.

chuwik commented on .devcontainer/devcontainer.json:
```diff
@@ -1,19 +1,31 @@
+    ],
+    // Use 'postCreateCommand' to run commands after the container is created.
+    "postCreateCommand": "cd app/frontend && npm install && cd ../backend && pip install -r requirements.txt"
+    // Uncomment to connect as root instead. More info: https://aka.ms/dev-containers-non-root.
```
Suggestion: if a remoteUser isn't required, let's remove it to keep the file as simple as possible.

iMicknl replied:
Addressed in b4af6e5d4bf617ad521bdab516b1f9e842254029

iMicknl commented:
>Would it better to run ./start.sh in postCreateCommand rather than just restoring the packages? That'd restore + build + run the app straight away. If you can test it and it works, I'd say go for it.

Starting the application directly after container creation will not be helpful, because the user will need add their environment variables and/or login via `az login` (AZ CLI). 

Addressed your feedback in b4af6e5d4bf617ad521bdab516b1f9e842254029! The 3.11 image works as well, tested with key and az cli credential.

chuwik commented:
Thanks @iMicknl, looking good and merged.

Out of curiosity, what's the difference between the mcr.microsoft.com/devcontainers/python:1-3.11-bookworm and mcr.microsoft.com/devcontainers/python:1-3.11 images?

iMicknl commented:
> Out of curiosity, what's the difference between the mcr.microsoft.com/devcontainers/python:1-3.11-bookworm and mcr.microsoft.com/devcontainers/python:1-3.11 images?

My preference is to pin the OS version (e.g. `bookworm` for the latest stable Debian release) to not have any side effects when there are new Debian releases. The default Devcontainer 3.11 [will refer to bookworm as well at the moment](https://github.com/devcontainers/images/blob/main/src/python/.devcontainer/Dockerfile). So at the moment there would be no difference in behavior.
